tests/tracer/test_encoders.py

```diff
@@ -520,27 +521,59 @@ def test_span_event_encoding_msgpack(version):
     with mock.patch("ddtrace._trace.span.time_ns", return_value=2234567890123456):
         span._add_event("We are going to the moon")
 
+    if version == "v0.4":
+        # test top level span event encoding
+        agent_config.trace_native_span_events = True
```

Comment:
I wonder if we should add additional parameters to the test function, e.g.

```
@pytest.mark.parametrize(
    "version,trace_native_span_events",
    [
        ("v0.4", False),
        ("v0.4", True),
        ("v0.5", False),
    ]
)
def test_span_event_encoding_msgpack(version, trace_native_span_events):
```


Since we probably don't want to lose the assertions with it both True and False?